src/lando/api/legacy/workers/uplift_worker.py

```diff
@@ -0,0 +1,235 @@
+import json
+import logging
+import os
+import subprocess
+import tempfile
+
+from typing_extensions import override
+
+from lando.api.legacy.workers.base import Worker
+from lando.main.models import (
+    JobAction,
+    PermanentFailureException,
+    Revision,
+    TemporaryFailureException,
+    WorkerType,
+)
+from lando.main.models.uplift import UpliftJob, UpliftRevision
+from lando.utils.tasks import (
+    send_uplift_failure_email,
+    send_uplift_success_email,
+    set_uplift_request_form_on_revision,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class UpliftWorker(Worker):
+    """Worker to execute uplift jobs.
+
+    This worker runs `UpliftJob`s on enabled repositories.
+    These jobs apply patches to respositories and create new Phabricator
+    revisions on success.
+    """
+
+    job_type = UpliftJob
+
+    worker_type = WorkerType.UPLIFT
+
+    @override
+    def refresh_active_repos(self):
+        """Override base functionality by not checking treestatus."""
+        self.active_repos = self.enabled_repos
+
+    @override
+    def run_job(self, job: UpliftJob) -> bool:
+        """Run an uplift job."""
+        repo = job.target_repo
+        user = job.multi_request.user
+        job_url = job.url()
+
+        try:
+            created_revision_ids = self.apply_and_uplift(job)
+        except TemporaryFailureException:
+            return False
+        except PermanentFailureException as exc:
+            self.notify_uplift_failure(job, repo.name, job_url, user.email, str(exc))
+            return False
+        except Exception as exc:  # pragma: no cover - defensive catch
+            self.notify_uplift_failure(job, repo.name, job_url, user.email, str(exc))
+            return False
+
+        self.notify_uplift_success(
+            repo.name,
+            job_url,
+            user.email,
+            created_revision_ids,
+        )
+        return True
+
+    def apply_and_uplift(self, job: UpliftJob) -> list[int]:
+        """Apply uplift patches to the repo and create new revisions.
+
+        Returns an ordered list of created revision IDs.
+        """
+        repo = job.target_repo
+        multi_request = job.multi_request
+        user = multi_request.user
+        scm = repo.scm
+
+        # Update to the latest commit in the target train.
+        base_revision = self.update_repo(repo, job, scm, target_cset=None)
+
+        # Apply patches to the tip of the target train.
+        def apply_patch(revision: Revision):
```

Comment:
nit: make the name more differentiable
```suggestion
        def log_apply_patch(revision: Revision):
```